pkg/reconciler/ksvc/resources/knative_configuration.go

```diff
@@ -26,61 +27,48 @@ func CreateModelServingContainer(modelName string, modelSpec *v1alpha1.ModelSpec
 	}
 }
 
-func CreateKnativeConfiguration(kfsvc *v1alpha1.KFService) (*knservingv1alpha1.Configuration, *knservingv1alpha1.Configuration) {
+func CreateKnativeConfiguration(kfsvc *v1alpha1.KFService, name string, modelSpec *v1alpha1.ModelSpec) *knservingv1alpha1.Configuration {
+	if modelSpec == nil {
+		return nil
+	}
 	annotations := make(map[string]string)
-	if kfsvc.Spec.MinReplicas != 0 {
-		annotations[autoscaling.MinScaleAnnotationKey] = fmt.Sprint(kfsvc.Spec.MinReplicas)
+	if modelSpec.MinReplicas != 0 {
+		annotations[autoscaling.MinScaleAnnotationKey] = fmt.Sprint(modelSpec.MinReplicas)
 	}
-	if kfsvc.Spec.MaxReplicas != 0 {
-		annotations[autoscaling.MaxScaleAnnotationKey] = fmt.Sprint(kfsvc.Spec.MaxReplicas)
+	if modelSpec.MaxReplicas != 0 {
+		annotations[autoscaling.MaxScaleAnnotationKey] = fmt.Sprint(modelSpec.MaxReplicas)
 	}
 
-	defaultConfiguration := &knservingv1alpha1.Configuration{
+	//Default to 1 for now we may need to expose this field
+	annotations[autoscaling.TargetAnnotationKey] = constants.DefaultScalingTarget
```

Comment:
maybe for some of these more advanced options, we don't model them as properties on our spec, but if the user sets the annotation manually we respect that and don't override it? That would imply you check before setting the value?

Reply:
SGTM, it is also an annotation in knative so user can just pass down the annotation then.